src/Compilers/VisualBasic/Test/Semantic/Semantics/RequiredMembersTests.vb

```diff
@@ -0,0 +1,1842 @@
+        Public Sub GenericConstrainedToNew_Forbidden(typeKind As String)
+            Dim cComp = CreateCSharpCompilationWithRequiredMembers(GetCDefinition(hasSetsRequiredMembers:=False, typeKind))
+
+            Dim vbCode = "
+Module M
+    Sub Main()
+        Generic(Of C)()
+    End Sub
+
+    Sub Generic(Of T As New)()
+    End Sub
+End Module"
+
+            Dim comp = CreateCompilation(vbCode, {cComp.EmitToImageReference()})
+            comp.AssertTheseDiagnostics(<expected>
+BC37324: 'C' cannot satisfy the 'New' constraint on parameter 'T' in the generic type or or method 'Public Sub Generic(Of T As New)()' because 'C' has required members.
+        Generic(Of C)()
+        ~~~~~~~~~~~~~
+                                        </expected>)
+        End Sub
+
+        <Theory>
+        <InlineData("class")>
+        <InlineData("struct")>
+        Public Sub GenericConstrainedToNew_HasSetsRequiredMembers_Allowed(typeKind As String)
+            Dim cComp = CreateCSharpCompilationWithRequiredMembers(GetCDefinition(hasSetsRequiredMembers:=True, typeKind))
+
+            Dim vbCode = "
+Module M
+    Sub Main()
+        Generic(Of C)()
+    End Sub
+
+    Sub Generic(Of T As New)()
+    End Sub
+End Module"
+
+            Dim comp = CreateCompilation(vbCode, {cComp.EmitToImageReference()})
+            comp.AssertNoDiagnostics()
+        End Sub
+
+        <Fact>
+        Public Sub GenericSubstitution_NoneSet()
+            Dim cDef = "
+public class C<T>
+{
+    public required T Prop { get; set; }
+    public required T Field;
+}"
+
+            Dim cComp = CreateCSharpCompilationWithRequiredMembers(cDef)
+
+            Dim vbCode = "
+Module M
+    Sub Main()
+        Dim c = New C(Of Integer)()
+    End Sub
+End Module"
+
+            Dim comp = CreateCompilation(vbCode, {cComp.EmitToImageReference()})
+            comp.AssertTheseDiagnostics(<expected>
+BC37321: Required member 'Public Field As Integer' must be set in the object initializer or attribute arguments.
+        Dim c = New C(Of Integer)()
+                    ~~~~~~~~~~~~~
+BC37321: Required member 'Public Overloads Property Prop As Integer' must be set in the object initializer or attribute arguments.
+        Dim c = New C(Of Integer)()
+                    ~~~~~~~~~~~~~
+                                        </expected>)
+        End Sub
+
+        <Fact>
+        Public Sub GenericSubstitution_AllSet()
+            Dim cDef = "
+public class C<T>
+{
+    public required T Prop { get; set; }
+    public required T Field;
+}"
+
+            Dim cComp = CreateCSharpCompilationWithRequiredMembers(cDef)
+
+            Dim vbCode = "
+Module M
+    Sub Main()
+        Dim c = New C(Of Integer)() With { .Prop = 1, .Field = 2 }
+    End Sub
+End Module"
+
+            Dim comp = CreateCompilation(vbCode, {cComp.EmitToImageReference()})
+            comp.AssertNoDiagnostics()
+        End Sub
+
+        <Fact>
+        Public Sub GenericSubstitution_Unbound()
```

Comment:
>Unbound [](http://example.com/codeflow?start=39&length=7)

Does this test really cover ```UnboundGenericType``` symbol? Probably not, because ```UnboundGenericType``` shouldn't have any members.